utils/src/time.rs

```diff
@@ -0,0 +1,73 @@
+//! Utility functions for `std::time`.
+
+use std::time::{Duration, SystemTime};
+
+/// Extension trait to add methods to `std::time::SystemTime`
+pub trait SystemTimeExt {
```

Comment:
I thought this was the best name (similar to `StreamExt`) but lmk if any preference. Maybe `TimeExt`?

It's published directly in the `utils` crate so the `time` part of the name shouldn't be redundant

Reply:
I think the name is great

Reply:
I agree with making it directly public rather than `time::Ext`